src/core/lib/compression/message_compress.cc

```diff
@@ -34,7 +34,7 @@
 static int zlib_body(z_stream* zs, grpc_slice_buffer* input,
                      grpc_slice_buffer* output,
                      int (*flate)(z_stream* zs, int flush)) {
-  int r;
+  int r = Z_OK;
```

Comment:
I wonder if it should be initialized as Z_STREAM_END instead. Otherwise compressing an empty string will end up in a failure?

Reply:
yes, and I think that should be right, since without the correct headers and footers, it makes no sense to call the message decompression a success

Reply:
thinking again, Z_STREAM_END should also be fine, since the compression/decompression semantics are in our control, and it would may be fine to call it a success on an empty message

Reply:
done